manage

```diff
@@ -0,0 +1,592 @@
+#!/usr/bin/env bash
+# -*- coding: utf-8; mode: sh indent-tabs-mode: nil -*-
+# SPDX-License-Identifier: AGPL-3.0-or-later
+# shellcheck disable=SC2031
+
+# shellcheck source=utils/lib.sh
+source "$(dirname "${BASH_SOURCE[0]}")/utils/lib.sh"
+# shellcheck source=utils/brand.env
+source "${REPO_ROOT}/utils/brand.env"
+source_dot_config
+
+# config
+
+export PY_SETUP_EXTRAS='[test]'
+export NPM_PACKAGES="less@2.7 less-plugin-clean-css grunt-cli"
+export GECKODRIVER_VERSION="v0.28.0"
+export GH_PAGES="build/gh-pages"
+export DOCS_DIST="${LXC_ENV_FOLDER}dist/docs"
+export DOCS_BUILD="${LXC_ENV_FOLDER}build/docs"
+
+if [ "$VERBOSE" = "1" ]; then
+    SPHINX_VERBOSE="-v"
+    PYLINT_VERBOSE="-v"
+fi
+
+# SPHINXOPTS=
+
+PYLINT_SEARX_DISABLE_OPTION="\
+I,C,R,\
+W0105,W0212,W0511,W0603,W0613,W0621,W0702,W0703,W1401,\
+E1136"
+
+PYLINT_ADDITIONAL_BUILTINS_FOR_ENGINES="supported_languages,language_aliases"
+PYLINT_OPTIONS="-m pylint -j 0 --rcfile .pylintrc"
+
+# These py files are linted by test.pylint(), all other files are linted by
+# test.pep8()
+PYLINT_FILES=(
+    searx/preferences.py
+    searx/testing.py
+    searx/engines/gigablast.py
+    searx/engines/deviantart.py
+    searx/engines/digg.py
+    searx/engines/google.py
+    searx/engines/google_news.py
+    searx/engines/google_videos.py
+    searx/engines/google_images.py
+    searx/engines/mediathekviewweb.py
+)
+
+pyenv.check() {
+    cat  <<EOF
+import yaml
+print('import yaml --> OK')
+EOF
+}
+
+ci.data() {
+    engines.languages
+    useragents.update
+    build_msg DATA "update searx/data/ahmia_blacklist.txt"
+    pyenv.cmd python utils/fetch_ahmia_blacklist.py
+    build_msg DATA "update searx/data/wikidata_units.json"
+    pyenv.cmd python utils/fetch_wikidata_units.py
+    # python utils/fetch_currencies.py
+}
+
+ci.themes() {
+    themes.oscar
+    themes.simple
+    themes.bootstrap
+}
+
+# shellcheck disable=SC2155
+
+ci.gh-pages() {
+
+    # The commit history in the gh-pages branch makes no sense, the history only
+    # inflates the repository unnecessarily.  Therefore a *new orphan* branch
+    # is created each time we deploy at gh-pages.
+
+    docs.clean
+    docs.prebuild
+    docs.html
+
+    [ "$VERBOSE" = "1" ] && set -x
+    local head="$(git rev-parse HEAD)"
+    local branch="$(git name-rev --name-only HEAD)"
```

Comment:
On branch master, this should return `master` but the CI returns `remotes/pull/2594/merge` which might be due to the PR cycle of github. I will look for an alternative.

Reply:
According to 
https://github.com/JamesIves/github-pages-deploy-action/tree/3.7.1#optional-choices

there is `SINGLE_COMMIT` option.

https://github.com/searx/searx/blob/5f4a085fc44ed64499d1aa5c2e29d2f861037b5f/.github/workflows/integration.yml#L77-L84

So
```yaml
    - name: Build documentation
      run: SEARX_DEBUG=1 make V=1 doc.html
    - name: Deploy
      if:  github.ref == 'refs/heads/master'
      uses: JamesIves/github-pages-deploy-action@3.7.1
      with:
        GITHUB_TOKEN: ${{ github.token }}
        BRANCH: gh-pages
        FOLDER: build/gh-pages
        CLEAN: true # Automatically remove deleted files from the deploy branch
		SINGLE_COMMIT: true
```

* On "Build documentation", the documentation is built. This always happens, on the master branch or in the PR.
* On "Deploy", the documentation is pushed to the gh-pages branch using `JamesIves/github-pages-deploy-action`. Only in the master branch.

Reply:
OK, what do you suggest .. should we use `JamesIves/github-pages-deploy-action` further?

BTW: I forgot to remove https://github.com/searx/searx/blob/5f4a085fc44ed64499d1aa5c2e29d2f861037b5f/.github/workflows/integration.yml#L77-L84

In general I am not a fan boy of (Github-) actions I can't reproduce locally, that was also one reason for me to change back to a (more simple) script which can be executed locally. 

Another reason for me is, that I know git and bash but Github-Actions are new to me / and to be true I don't want to investigate to much time in tools that can only be used on a proprietary platform.

Another point that disturbs me with, is the the fact that this github-Action needs hundred of lines of code for a simple commit.

But you are the maintainer and you should decide what is the best for you .. 

What do you think, should screw it back to the old, existing method where the scripts compile the docs and _Ives_ deploys? 

I think we can discuss my doubt in a separated issue ..

Reply:
FYI: I am working on a _rebase_ fixing this issue and some other issues I found.